test/bats/azure.bats

```diff
@@ -23,8 +24,19 @@ setup() {
   run helm install charts/secrets-store-csi-driver -n csi-secrets-store --namespace dev \
           --set image.pullPolicy="IfNotPresent" \
           --set image.repository="e2e/secrets-store-csi" \
-          --set image.tag=$IMAGE_TAG \
-          --set providers.azure.enabled=true
+          --set image.tag=$IMAGE_TAG
+  assert_success
+}
+
+@test "install azure provider" {
+  run kubectl apply -f $PROVIDER_YAML
```

Comment:
this will deploy the provider manifest in the default namespace. This will not be an issue - but maybe we should deploy all the components in the same namespace or maybe add a note here just so we are aware of it.

Reply:
Added $NAMESPACE to doc and e2e for vault and azure.